### PR TITLE
Update Node.js to 20.10.0 LTS and replace appdmg with macdmg

### PR DESCRIPTION
**Pull request by haslinghuis**

### CODE REVIEW

**Comment by haslinghuis** [comment]
Dependency hell :)

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Comment by McGiverGim** [comment]
It seems MAC build is failing?

**Comment by haslinghuis** [comment]
Already upgraded all dependencies and it works on Linux :)
Requires more work.

File: .nvmrc
Comment (McGiverGim) [inline .nvmrc:1]: I suppose is better to move to the latest LTS, that is 18.16.0?
```suggestion
18.16.0
```

**Comment by McGiverGim** [comment]
@haslinghuis in my case, upgrading:
```
"gulp-xml-transformer": "^7.0.0",
```
in the `package.json` fixes windows build.

**Comment by McGiverGim** [comment]
> @haslinghuis in my case, upgrading:
> 
> ```
> "gulp-xml-transformer": "^7.0.0",
> ```
> 
> in the `package.json` fixes windows build.

Sorry, fixes `yarn install`, but not windows build :( I will keep investigating.

**Review by McGiverGim — APPROVED** [review]

**Review by chmelevskij — APPROVED** [review]

**Comment by ctzsnooze** [comment]
This makes it very difficult on a mac.
It requires me to re-install rpm and that fails on my (older) mac.

**Comment by ctzsnooze** [comment]
Doesn't work, wasted two hours on this.  
Had to reinstall node, nvm, nothing worked.

**Comment by ctzsnooze** [comment]
This is a summary:
```
chris:betaflight-config-ctzsnooze chris$ node -v
v20.10.0
chris:betaflight-config-ctzsnooze chris$ nvm -v
0.37.2
chris:betaflight-config-ctzsnooze chris$ gulp
/Users/chris/Documents/Personal/RC/betaflight-config-ctzsnooze/gulpfile.js:11
const NwBuilder = require('nw-builder');
                  ^

Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/chris/Documents/Personal/RC/betaflight-config-ctzsnooze/node_modules/nw-builder/src/index.js from /Users/chris/Documents/Personal/RC/betaflight-config-ctzsnooze/gulpfile.js not supported.
Instead change the require of index.js in /Users/chris/Documents/Personal/RC/betaflight-config-ctzsnooze/gulpfile.js to a dynamic import() which is available in all CommonJS modules.
    at Object.<anonymous> (/Users/chris/Documents/Personal/RC/betaflight-config-ctzsnooze/gulpfile.js:11:19) {
  code: 'ERR_REQUIRE_ESM'
}

Node.js v20.10.0
chris:betaflight-config-ctzsnooze chris$ yarn start
yarn run v1.22.21
$ run-script-os
$ NODE_ENV=development NW_PRE_ARGS=--load-extension='./node_modules/nw-vue-devtools-prebuilt/extension' gulp debug
Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/chris/Documents/Personal/RC/betaflight-config-ctzsnooze/node_modules/nw-builder/src/index.js from /Users/chris/Documents/Personal/RC/betaflight-config-ctzsnooze/gulpfile.js not supported.
Instead change the require of index.js in /Users/chris/Documents/Personal/RC/betaflight-config-ctzsnooze/gulpfile.js to a dynamic import() which is available in all CommonJS modules.
    at Object.<anonymous> (/Users/chris/Documents/Personal/RC/betaflight-config-ctzsnooze/gulpfile.js:11:19) {
  code: 'ERR_REQUIRE_ESM'
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
chris:betaflight-config-ctzsnooze chris$ 
```

**Comment by ctzsnooze** [comment]
Well, don't ask me why... I tried to fix it by adding `"type": "module"` to package.json, and in gulpfile.js changing `const NwBuilder = require('nw-builder');` to `const NwBuilder = module ('nw-builder');` 
None of that worked, so I reverted those changes.
Then suddenly it started working!
These commands resulted in mac binaries...
- `yarn start` or`gulp` makes a directory tree `/debug/betaflight-configurator/osx64/`, and in `osx64/` the `betaflight-configurator.app` is locate (with other files).  The app works and supports developer tools.
- yarn release makes a directory `release/` containing a mag `dmg` file which, when double-clicked, mounts as a volume and contains the `betaflight-configurator.app`.  This file could be distributed 'as it is` without conversion to zip.

**Comment by chmelevskij** [comment]
tbh, haven't tried i myself. @haslinghuis did you verify it actually works or does it need more testing?

**Review by chmelevskij — CHANGES_REQUESTED** [review]
Sorry, doesn't work on apple silicon:

```
[17:55:55] '<anonymous>' errored after 4.04 s
[17:55:55] Error in plugin "gulp-macdmg"
Message:
    Command failed: bless --folder /Volumes/Betaflight Configurator --openfolder /Volumes/Betaflight Configurator
bless: The 'openfolder' is not supported on Apple Silicon devices.


Details:
    code: 1
    stdout:
    stderr: bless: The 'openfolder' is not supported on Apple Silicon devices.

    failed: true
    signal: null
    cmd: bless --folder /Volumes/Betaflight Configurator --openfolder /Volumes/Betaflight Configurator
    timedOut: false
    killed: false
    domainEmitter: [object Object]
    domainThrown: false
 ```
 
 Weirdly enough it worked for `start` development mode, but got the error on `release` 🤔

**Comment by haslinghuis** [comment]
Did you check if node.js is the arm64 (not x64) version on your Mac?

**Comment by ctzsnooze** [comment]
I have had issues with stalled debug saves, slow-downs when pasting a long CLI diff all, and stalled saved Preset imports, requiring a reboot of Configurator, since testing this PR on my mac.

Let's not go down this path until the dust has settled on 4.5.  Right now, at release time for 4.5, is not the best time to change these things, I think.

**Comment by haslinghuis** [comment]
Closing in favor of #3673

**Pull request closed without merging**
